src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ResourceDictionary.cs

```diff
@@ -1735,23 +1735,31 @@ private void NotifyOwners(ResourcesChangeInfo info)
                     DeferredResourceReference deferredResourceReference;
                     if (!IsThemeDictionary)
                     {
-                        if (_ownerApps != null)
-                        {
-                            deferredResourceReference = new DeferredAppResourceReference(this, resourceKey);
-                        }
-                        else
-                        {
-                            deferredResourceReference = new DeferredResourceReference(this, resourceKey);
-                        }
-
                         // Cache the deferredResourceReference so that it can be validated
                         // in case of a dictionary change prior to its inflation
                         if (_deferredResourceReferences == null)
                         {
-                            _deferredResourceReferences = new WeakReferenceList();
+                            _deferredResourceReferences = new DeferredResourceReferenceList();
+                        }
+
+                        if (_deferredResourceReferences.Get(resourceKey) is DeferredResourceReference existingDeferredResourceReference
```

Comment:
This line results in Type check and null check both. As Type is well known, only null check shall be enough. 
Like
`
var existingDeferredResourceReference = _deferredResourceReferences.Get(resourceKey);
if (existingDeferredResourceReference?.Dictionary == this)
`

Reply:
Fixed by using an inline check via `is { }`.